app/src/ui/tab-bar.tsx

```diff
@@ -88,6 +96,31 @@ export class TabBar extends React.Component<ITabBarProps, {}> {
     }
   }
 
+  /**
+   * If something is being dragged, this allows for tab selection by hovering
+   * over a tab for a few seconds (dragTabSwitchWaitTime)
+   */
+  private onMouseEnter = (index: number) => {
+    if (
+      index === this.props.selectedIndex ||
+      !dragAndDropManager.isDragInProgress ||
+      this.props.allowDragOverSwitching === undefined ||
```

Comment:
I wonder if this should be `true` by default. That way the user could learn they can switch between tabs from anywhere. If I dragged the commits from the History tab over to the Changes tab and it didn't work, maybe I wouldn't try to do the same inside the branch dropdown.

What do you think?

Reply:
:) I had the exact opposite thought. I originally had it open for anywhere, but then I was like.. we would not want the user to be able to navigate somewhere that they can't do anything with commits (Changes tab) (similar concept to instituting the drag overlay - narrow the users possibilities to their target). However, to your point... I wondered if it would make any sense to style the tabs differently to indicate dragging over them does something... or highlight them when you drag over them.. but that may feel like you can drop on them.

@ampinsk thoughts?

Reply:
I think I agree with @tidy-dev on this one. There's nothing for you to do in the "Changes" tab, so you shouldn't be able to drag commits there. I think it would be jarring and unexpected to end up there, and one of our core tenets of Desktop is to help people get to where they need to go and avoid unintended mishaps.

Reply:
@billygriffin I totally agree with that. My concern is more about the learning process: trying to switch to the "Changes" tab while you're dragging commits (because at that point you don't know where to drop them), you see it doesn't work but don't know why, so you might think that's something you can't do anywhere in the app. Specially since you need to keep the mouse over the element for a few milliseconds: if you see it doesn't work in one place and you're not patient enough when you try in the Branches/PRs tab in the dropdown, you might think it doesn't work there either.

IMHO, it also makes the UI feel inconsistent (in terms of behavior): I can't switch to "Changes" because there is nothing for me to do there, but I can switch from "Branches" to "Pull Requests"? (that's how the current implementation works, if I read it correctly)

> I wondered if it would make any sense to style the tabs differently to indicate dragging over them does something... or highlight them when you drag over them.. but that may feel like you can drop on them.

I think _any_ kind of positive/negative feedback would work: if we disable switching tabs to "Changes" but we somehow inform the user, they might understand it's something specific to that tab.